src/components/[guild]/crm/useMembers.tsx

```diff
@@ -0,0 +1,93 @@
+import { useCallback, useMemo, useState } from "react"
+import useSWRInfinite from "swr/infinite"
+import { PlatformAccountDetails, Visibility } from "types"
+import { useFetcherWithSign } from "utils/fetcher"
+import useGuild from "../hooks/useGuild"
+
+type CrmRole = {
+  roleId?: number
+  requirementId?: number
+  access?: boolean
+  amount?: number
+}
+
+type Member = {
+  userId: number
+  addresses: string[]
+  platformUsers: PlatformAccountDetails[]
+  joinedAt: string
+  roles: {
+    hidden?: CrmRole[]
+    public: CrmRole[]
+  }
+}
+
+const LIMIT = 50
+
+const useMembers = (queryString) => {
+  const { id } = useGuild()
+  const [hasReachedTheEnd, setHasReachedTheEnd] = useState(true)
+
+  const getKey = useCallback(
```

Comment:
We should return null if the user's keyPair isn't ready, so we don't get a signing request popup on initial load, I did a similar thing inside `ActivityLogContext`, it should be a simple copy-paste from there!